sycl/include/sycl/detail/defines.hpp

```diff
@@ -27,7 +27,8 @@
 #define __SYCL_SPECIAL_CLASS
 #endif
 
-#if __has_cpp_attribute(__sycl_detail__::sycl_type)
+#if __has_cpp_attribute(__sycl_detail__::sycl_type) &&                         \
```

Comment:
Can you modify definition of SYCLTYPE to use let LangOpts = [SYCLIsDevice, SilentlyIgnoreSYCLIsHost];